carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/ParticipantServiceHost.kt

```diff
@@ -0,0 +1,180 @@
+package dk.cachet.carp.studies.application
+
+import dk.cachet.carp.common.EmailAddress
+import dk.cachet.carp.common.UUID
+import dk.cachet.carp.common.users.AccountIdentity
+import dk.cachet.carp.common.users.EmailAccountIdentity
+import dk.cachet.carp.deployment.application.DeploymentService
+import dk.cachet.carp.deployment.domain.StudyDeploymentStatus
+import dk.cachet.carp.studies.domain.ParticipantGroupStatus
+import dk.cachet.carp.studies.domain.Study
+import dk.cachet.carp.studies.domain.StudyRepository
+import dk.cachet.carp.studies.domain.users.AssignParticipantDevices
+import dk.cachet.carp.studies.domain.users.DeanonymizedParticipation
+import dk.cachet.carp.studies.domain.users.Participant
+import dk.cachet.carp.studies.domain.users.ParticipantRepository
+import dk.cachet.carp.studies.domain.users.deviceRoles
+import dk.cachet.carp.studies.domain.users.participantIds
+
+
+class ParticipantServiceHost(
```

Comment:
Nothing really new here. These are just methods moved from `StudyService` to this new interface.

Most relevant is to note that this service depends on `StudyRepository` as well to verify whether the study exists, and to add participations when deploying participant groups.